src/main/java/core/basesyntax/strategy/SupplyOperation.java

```
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> stock) {
```

This handler does not validate null parameters. The project guidance explicitly lists null-parameter validation as required: "Null parameters (e.g., null file path, null list inputs) should be validated and cause informative exceptions." Please add checks for `transaction` and `stock` (and for `transaction.getFruit()` if applicable) and throw an informative RuntimeException when any are null instead of allowing a NullPointerException deeper in the code. See the project description for null-parameter requirements and the interface signature for the handler.